src/install/lockfile/Tree.zig

```diff
@@ -256,6 +256,19 @@ pub fn Builder(comptime method: BuilderMethod) type {
         install_root_dependencies: if (method == .filter) bool else void,
         packages_to_install: if (method == .filter) ?[]const PackageID else void,
 
+        pub const FillItem = struct {
+            tree_id: Tree.Id,
+            dependency_id: DependencyID,
+
+            /// If valid, dependencies will not hoist
+            /// beyond this tree if they're in a subtree
+            hoist_root_id: Tree.Id,
+
+            subpath: if (method == .filter) bun.collections.ArrayListDefault(u8) else void,
+        };
```

Comment:
Would add a docblock explaining what the intent behind this is. It's kind of unclear at a glance